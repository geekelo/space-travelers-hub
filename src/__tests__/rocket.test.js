import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import Rocket from '../components/rocket/rocket';

const mockStore = configureMockStore();
const initialState = {
  rocket: {
    value: [
      {
        id: 1,
        name: 'Falcon 1',
        description: 'Description for Falcon 1',
        flickr_images: ['image1.jpg', 'image2.jpg'],
        active: false,
      },
      // ... Add more rockets as needed for testing
    ],
    status: 'done',
    error: false,
  },
};
const store = mockStore(initialState);

describe('Rocket Component', () => {
  it('matches the snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <Rocket />
      </Provider>,
    );

    // Take a snapshot of the rendered output
    expect(container).toMatchSnapshot();
  });
});
