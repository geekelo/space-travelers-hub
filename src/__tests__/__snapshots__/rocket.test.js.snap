// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rocket Component matches the snapshot 1`] = `
<div>
  <div>
    <ul>
      <li
        class="cards"
      >
        <div>
          <img
            alt="Rocket"
            src="image1.jpg"
            width="200"
          />
        </div>
        <div
          class="contentSection"
        >
          <h2>
            Falcon 1
          </h2>
          <p
            class="description"
          >
            Description for Falcon 1
          </p>
          <p>
            <button
              type="submit"
            >
              Reserve Rocket
            </button>
          </p>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
